packages/navigator/src/browser/navigator-preferences.ts

```diff
@@ -24,12 +24,19 @@ export const FileNavigatorConfigSchema: PreferenceSchema = {
             type: 'boolean',
             description: 'Selects file under editing in the explorer.',
             default: true
+        },
+        'explorer.compactFolders': {
+            type: 'boolean',
+            description: `Controls whether the explorer should render folders in a compact form. In such a form, single child folders will be compressed in a combined tree element.
```

Comment:
Please localize the following preference description:

```suggestion
            // eslint-disable-next-line max-len
            description: nls.localizeByDefault('Controls whether the explorer should render folders in a compact form. In such a form, single child folders will be compressed in a combined tree element. Useful for Java package structures, for example.'),
```